Add document explaining ignition-template (#2102)
cybozu-go/neco: docs/ignition-template.md

Ignition-template
=========

> [ignition-template](../ignition-template) automatically generates configuration files for each environment such as `common-<cluster>.yml` and `site-<cluster>.yml`.

Reviewer: It would be good to place some introduction about [Ignition template system](https://github.com/cybozu-go/sabakan/blob/master/docs/ignition_template.md).
You can just point to `docs/ignition.md`.

This document should be pointed from `docs/ignition.md`.

> [ignition-template](../ignition-template) automatically generates configuration files for each environment such as `common-<cluster>.yml` and `site-<cluster>.yml`.

Author: The proper use of the terms was not being used, so the config file are replaced with two terms: ignition template and pseudo-ignition.
b92b78602c1aa7bb6494cfcb4722d07551c5a69e

Also add reference to documentation.
a0915aa6478e333823b97650ba3f740504535691

It can also generate configuration files such as `site.yml`.
The configuration files are generated based on [settings.json](../ignition-template/settings.json).
If you change the contents of the configuration files or add new configuration files, please modify the [settings.json](../ignition-template/settings.json) file.

A configuration file can be generated by the following command.
```bash
make setup
# Generate and place site.yml|site-<cluster>.yml files.
make site-env
# Generate and place config.yml|config-<cluster>.yml files.
make common-env
```

## Generate diffs of configurations for each environment

A configgenerator command is provided to examine the differences between each environment.
The configgenerator generates an configuration file based on the specified role and cluster.

> Configgenerator command cluster are optional and will work even if only role is specified in the configgenerator command.

Reviewer: ```suggestion
Configgenerator command takes a cluster as an option and will work even if only the role is specified.
```

Author: Fix in 1a4505f90eeaf3c500a05938b796de6770092b99
